serenity-gradle-plugin: task checkOutcomes causes java.lang.StackOverflowError

Hello

### Info:
serenity-gradle-plugin version:2.0.48
gradle version: http\://services.gradle.org/distributions/gradle-2.12-all.zip
OS: macOS Mojave 10.14.2

### Issue:
When I run `./gradlew checkOutcomes --stacktrace`
```
FAILURE: Build failed with an exception.

* What went wrong:
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
java.lang.StackOverflowError
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:82)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:83)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:83)

...etc

BUILD FAILED
```

I copied the checkOutcomes task's code from [SerenityPlugin.groovy](https://github.com/serenity-bdd/serenity-core/blob/master/serenity-gradle-plugin/src/main/groovy/net/serenitybdd/plugins/gradle/SerenityPlugin.groovy) and tried this locally and managed to identify the line 
```
inputs.files reportDirectory
```
causing the issue.
I printed the `reportDirectory` def and this points to {projectFolder}/target/site/serenity (/Users/dekeyzer/git/SLIMS6.1/slims-acceptance/target/site/serenity) which is normal.
I then thought it was because of the structure inside this directory but emptying it caused the same error.

I started looking at the [gradle documentation](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:up_to_date_checks) and saw they used 
```
inputs.files(fileTree("src/templates"))
```
in their examples so I tried this with
```
inputs.files(fileTree(reportDirectory))
```
and this solved my problem.

Is there something I'm missing or could this be a bug?

--- a/serenity-gradle-plugin/src/main/groovy/net/serenitybdd/plugins/gradle/SerenityPlugin.groovy
+++ b/serenity-gradle-plugin/src/main/groovy/net/serenitybdd/plugins/gradle/SerenityPlugin.groovy
@@ -90,7 +90,7 @@ class SerenityPlugin implements Plugin<Project> {
 
             log.info("SerenityPlugin:checkOutcomes: reportDirectory = ${reportDirectory}")
 
-            inputs.files reportDirectory
+            inputs.files(project.fileTree(reportDirectory))
 
             doLast {
                 updateProperties(project)

